cpu/cc2538/periph/gpio.c

```diff
@@ -26,19 +26,27 @@
 #include "periph/gpio.h"
 
 #define GPIO_MASK           (0xfffff000)
-#define PORTNUM_MASK        (0x00007000)
+#define PORTNUM_MASK        (0x00003000)
```

Comment:
Just tested it. That fix seems to be a remnant of a previous fix, where the 'PORTNUM_MASK' would give me negative numbers, so I reworked it. But the calculation and the 'PORTNUM_MASK' can remain the same for me, as long as the if-clause is present, in order to work with the old encoding.